rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -103,46 +99,7 @@ void SequentialCompressionWriter::open(
   const rosbag2_cpp::StorageOptions & storage_options,
   const rosbag2_cpp::ConverterOptions & converter_options)
 {
-  max_bagfile_size_ = storage_options.max_bagfile_size;
-  base_folder_ = storage_options.uri;
-
-  if (converter_options.output_serialization_format !=
-    converter_options.input_serialization_format)
-  {
-    converter_ = std::make_unique<rosbag2_cpp::Converter>(converter_options, converter_factory_);
-  }
-
-  rcpputils::fs::path db_path(base_folder_);
-  if (db_path.is_directory()) {
-    std::stringstream error;
-    error << "Database directory already exists (" << db_path.string() <<
-      "), can't overwrite existing database";
-    throw std::runtime_error{error.str()};
-  }
-
-  bool dir_created = rcpputils::fs::create_directories(db_path);
-  if (!dir_created) {
-    std::stringstream error;
-    error << "Failed to create database directory (" << db_path.string() << ").";
-    throw std::runtime_error{error.str()};
-  }
-
-  const auto storage_uri = format_storage_uri(base_folder_, 0);
-  storage_ = storage_factory_->open_read_write(storage_uri, storage_options.storage_id);
-  if (!storage_) {
-    throw std::runtime_error{"No storage could be initialized. Abort"};
-  }
-
-  if (max_bagfile_size_ != 0 &&
-    max_bagfile_size_ < storage_->get_minimum_split_file_size())
-  {
-    std::stringstream error;
-    error << "Invalid bag splitting size given. Please provide a value greater than " <<
-      storage_->get_minimum_split_file_size() << ". Specified value of " <<
-      storage_options.max_bagfile_size;
-    throw std::runtime_error{error.str()};
-  }
-
+  prepare_to_open(storage_options, converter_options);
```

Comment:
can this method be deleted by overriding `prepare_to_open`, calling the superclass, then calling `setup_compression`?

Reply:
I don't particularly see the difference except in that `setup_compression` is maybe part of the `prepare` stage of this? I don't have a problem implementing this but am interested to know why you think this is better.

Reply:
+1 to @emersonknapp's comment. This leaves the superclass alone and implements the compression specific details in the relevant place.